instrumentation/reactor-netty-0.9/src/main/java/io/opentelemetry/javaagent/instrumentation/reactor_netty/ReactorNettyInstrumentationModule.java

```diff
@@ -0,0 +1,112 @@
+      ReactorNettyInstrumentationModule.class.getName() + "$MapConnect",
+      ReactorNettyInstrumentationModule.class.getName() + "$OnRequest",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.AttributeKeys",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.AttributeKeys$1",
+      // these below a transitive dependencies of AttributeKeys from above
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.NettyHttpClientTracer",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.NettyResponseInjectAdapter",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.HttpClientRequestTracingHandler",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.HttpClientResponseTracingHandler",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.HttpClientTracingHandler",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.NettyHttpServerTracer",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.NettyRequestExtractAdapter",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.HttpServerRequestTracingHandler",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.HttpServerResponseTracingHandler",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.HttpServerTracingHandler"
+    };
+  }
+
+  @Override
+  public List<TypeInstrumentation> typeInstrumentations() {
+    return singletonList(new HttpClientInstrumentation());
+  }
+
+  private static final class HttpClientInstrumentation implements TypeInstrumentation {
+    @Override
+    public ElementMatcher<TypeDescription> typeMatcher() {
+      return named("reactor.netty.http.client.HttpClient");
+    }
+
+    @Override
+    public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+      return singletonMap(
+          isStatic().and(named("create")),
+          ReactorNettyInstrumentationModule.class.getName() + "$CreateAdvice");
+    }
+  }
+
+  public static class CreateAdvice {
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void onEnter() {
+      CallDepthThreadLocalMap.incrementCallDepth(HttpClient.class);
+    }
+
+    @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
+    public static void stopSpan(
+        @Advice.Thrown Throwable throwable, @Advice.Return(readOnly = false) HttpClient client) {
+
+      if (throwable == null && CallDepthThreadLocalMap.decrementCallDepth(HttpClient.class) == 0) {
+        client = client.doOnRequest(new OnRequest()).mapConnect(new MapConnect());
+      }
+    }
+  }
+
+  public static class MapConnect
+      implements BiFunction<Mono<? extends Connection>, Bootstrap, Mono<? extends Connection>> {
+    @Override
+    public Mono<? extends Connection> apply(Mono<? extends Connection> m, Bootstrap b) {
+      return m.subscriberContext(s -> s.put("otel_context", Context.current()));
```

Comment:
is this attribute name shared with any other instrumentation? if not, maybe name it `reactor_netty_otel_context`?

Reply:
Why, what difference does it make apart having longer key? :)

Reply:
when I saw this, the general name gave me the impression that this (reactor) context key was being used to propagate the (otel) context more generally for all reactor instrumentation

Reply:
Weeeeellll. That may be totally not a bad idea...